docs/how-to-guides/new-user-guides/kubernetes-clusters-in-rancher-setup/set-up-cloud-providers/amazon.md

```diff
@@ -159,6 +163,357 @@ Setting the value of the tag to `owned` tells the cluster that all resources wit
+
+```
+
+```
+
+### Migrating to the Out-of-Tree AWS Cloud Provider for RKE1
+
+In order to upgrade existing cluster with in-tree cloud provider to AWS cloud controller manager, you can run stop kube controller manager and install AWS cloud controller manager in many ways. Refer to [External cloud controller manager](https://cloud-provider-aws.sigs.k8s.io/getting_started/) for details.
+
+When downtime is acceptable, you can switch to external cloud provider which removes in-tree components and then deploy charts to install AWS cloud controller manager as explained in [using out of tree cloud provider](#using-out-of-tree-aws-cloud-provider).
+
+When control plane cannot tolerate downtime, leader migration must be enabled to facilitate a smooth migration from the controllers in the kube controller manager to their counterparts in the cloud controller manager. Refer to [Using leader migration](https://cloud-provider-aws.sigs.k8s.io/getting_started/) for more details.
+
+**Important**
+- [Cloud controller migration docs](https://kubernetes.io/docs/tasks/administer-cluster/controller-manager-leader-migration/#before-you-begin) mentions that it is possible to migrate with the same Kubernetes version, but assumes that migration is part of Kubernetes upgrade.
+
+- Refer [Migrate to use cloud controller manager](https://kubernetes.io/docs/tasks/administer-cluster/controller-manager-leader-migration/) to confirm if any customizations are required before migrating.
+  Confirm [migration configuration values](https://kubernetes.io/docs/tasks/administer-cluster/controller-manager-leader-migration/#default-configuration) and special case around [migrating IPAM controllers](https://kubernetes.io/docs/tasks/administer-cluster/controller-manager-leader-migration/#node-ipam-controller-migration).
+
+
+1. Update cluster config to enable leader migration in `cluster.yml`
+
+```
+services:
+  kube-controller:
+    extra_args:
+      enable-leader-migration: "true"
+```
+Note that cloud provider is still `aws` at this step.
+```
+cloud_provider:
+  name: aws
+```
+
+2. Cordon control plane nodes so aws cloud controller pods run on nodes only after upgrading to external cloud provider.
+
+```
+kubectl cordon -l "node-role.kubernetes.io/controlplane=true"
+```
+
+3. The steps to install AWS cloud controller manager for migration are same as fresh installation, but you must enable leader migration. 
+Add the following to the container arguments in step 7 when following the [steps to install chart](#helm-chart-installation):
+
+```
+- '--enable-leader-migration=true' 
+```
+
+4. Confirm the chart is installed but the new pods aren't running yet due to cordoned controlplane nodes. After updating the cluster in thenext step, rke will uncordon each node after upgrading and `aws-controller-manager` pods will be scheduled.
```

Comment:
```suggestion
4. Confirm the chart is installed but the new pods aren't running yet due to cordoned controlplane nodes. After updating the cluster in the next step, RKE uncordons each node after upgrading and schedules `aws-controller-manager` pods.
```

Reply:
@martyav I am a bit confused for this change, the uncordoning of node will happen only after updating cluster and upgrading nodes so shouldn't it be `will uncordon`?

Reply:
Maybe a little overzealous in changing tense. We can keep in "will" if that helps with clarity.